src/charm.py

```diff
@@ -652,24 +652,26 @@ def _update(self, event):
         pebble_layer = {
             "summary": "temporal server layer",
             "services": {
-                self.name: {
+                "temporal-server": {
```

Comment:
In the past, there used to be an issue with pebble; when refreshing a charm, if there was a new service with a different name, the container of the previous service wouldn't get removed and would linger around until the app was removed. Did you make sure this was not happening when refreshing the charm from latest to this PR (I know there are automated tests for the refresh, but it's hard to see this behaviour in the CI)?

Reply:
Confirmed by running upgrade from `latest/edge` to locally built charm

When the app is refreshed, juju (at least in 3.6) patches the statefulset spec with the new image. With this change, k8s removes old pods and creates new ones. With the newly created pods, there is no hint of the old service name as the new pod contains no reference of the old service name. This was confirmed by upgrading and then ensuring the new pod's pebble plan.

Reply:
Amazing, thanks!